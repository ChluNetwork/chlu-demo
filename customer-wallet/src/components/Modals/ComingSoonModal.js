import React from 'react'
import { bool, func } from 'prop-types'
import Dialog from '@material-ui/core/Dialog';
import Button from '@material-ui/core/Button'

const ComingSoon = ({ isOpen, hideModal }) => {
  const actions = [
    <Button
      label='OK'
      onClick={hideModal}
      primary
    />
  ]

  return (
    <div>
      <Dialog
        title='Work in progress'
        actions={actions}
        open={isOpen}
        onRequestClose={hideModal}
      >Currently we only support Bitcoin, all the other currencies shown here are coming soon.</Dialog>
    </div>
  )
}

ComingSoon.defaultProps = {
  isOpen: false
}

ComingSoon.propTypes = {
  isOpen: bool,
  hideModal: func
}

export default ComingSoon
