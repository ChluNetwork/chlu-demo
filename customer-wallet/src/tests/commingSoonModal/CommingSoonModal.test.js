import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ComingSoonModal from 'components/ComingSoon'

describe('<ComingSoonModal />', () => {

  it('should render correctly modal component', () => {
    const wrapper = shallow(
      <ComingSoonModal
        open={false}
        closeModal={() => null}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
