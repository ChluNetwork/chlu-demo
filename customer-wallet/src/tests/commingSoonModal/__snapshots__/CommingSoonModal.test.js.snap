// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComingSoonModal /> should render correctly modal component 1`] = `
<div>
  <Dialog
    actions={
      Array [
        <FlatButton
          disabled={false}
          fullWidth={false}
          label="OK"
          labelPosition="after"
          labelStyle={Object {}}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchStart={[Function]}
          primary={true}
          secondary={false}
      />,
      ]
    }
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
    title="Work in progress"
  >
    Currently we only support Bitcoin, all the other currencies shown here are coming soon.
  </Dialog>
</div>
`;
